lib/src/main/java/graphql/nadel/engine/transform/query/NadelQueryTransformer.kt

```diff
@@ -142,18 +147,23 @@ class NadelQueryTransformer private constructor(
             .build()
             .also { newField ->
                 // Track overall -> underlying fields
-                transformContext.overallToUnderlyingFields.compute(field) { _, oldValue ->
-                    (oldValue ?: emptyList()) + newField
-                }
+                transformContext.overallToUnderlyingFields
+                    .computeIfAbsent(field) {
+                        mutableListOf()
+                    }
+                    .also {
+                        it.add(newField)
+                    }
             }
     }
 
     private suspend fun applyTransformationSteps(
         field: ExecutableNormalizedField,
         transformationSteps: List<NadelExecutionPlan.Step<Any>>,
     ): NadelTransformFieldResult {
-        var fieldFromPreviousTransform: ExecutableNormalizedField = field
-        var aggregatedTransformResult: NadelTransformFieldResult? = null
+        var newField: ExecutableNormalizedField = field
+        val artificialFields = mutableListOf<ExecutableNormalizedField>()
```

Comment:
Add artificial fields to this mutable list intead of creating a new list every time